### ISSUE TITLE
N1LoaderReflection interface mismatch with ActiveRecord::Reflection::MacroReflection

### ISSUE THREAD

**Issue by senid231** [djezzzl/n1_loader#35]
when Project has buller gem (https://github.com/flyerhzm/bullet) following exception appear

```
# <NoMethodError>: undefined method `name'
# for #<struct N1Loader::ActiveRecord::Associations::Preloader::N1LoaderReflection key=..., loader=...>
# at
# records.each { |record| Bullet::Detector::Association.add_object_associations(record, reflection.name) }
# /.../bullet-7.0.7/lib/bullet/active_record61.rb:80:in `block in preloaders_for_reflection'
```

I've solved it like this
```ruby
N1Loader::ActiveRecord::Associations::Preloader::N1LoaderReflection.alias_method :name, :key
```

but I think `N1LoaderReflection` should have same public interface as `ActiveRecord::Reflection::MacroReflection` has,
to be safe from other problems that may arise in future

**Comment by djezzzl** [comment]
Hi @senid231 ,

Thank you for using the gem and raising the issue! I'm sorry it took me long to respond. 

Sounds good to me! Would you mind implementing it and contributing? I will be happy to merge and release it. Please, let me know if you can't, so I'll allocate time to implement it myself.

**Comment by djezzzl** [comment]
Hi @senid231,

Thank you for creating the issue, and I'm sorry it took me long to fix it.

It was done here: https://github.com/djezzzl/n1_loader/pull/42.

And it was released in 1.7.1. 

Please let me know if it helps.

Thank you, and have a great day!
